### ISSUE TITLE
web pages: All segments at our HTML specifications for Checks should have link

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#586]
We need to make all segments(first priority is link to Check, but ideally links to configs and examples .....) on HTML pages like 
http://checkstyle.sourceforge.net/config_whitespace.html

to be linkable to easy copy link and share it with others, to exactly point user to place to read.

example : http://checkstyle.sourceforge.net/config_whitespace.html#EmptyForInitializerPad

Good example of how it is done on other web sites:
http://logback.qos.ch/manual/configuration.html#statusListener and move mouse over a link text.
https://google-styleguide.googlecode.com/svn/trunk/javaguide.html#s1.2-guide-notes see at persistent image of chain-link.

**Comment by alexkravin** [comment]
@romani 
I made linkable sections at one page for testing approach, please review:
http://alexkravin.github.io/anchors/config_whitespace.html#EmptyForInitializerPad

As xdoc has very tremulous attitude to position of elements I have no possibility to insert the image-link near section name, so it's located under the title.

**Comment by romani** [comment]
hmm, that is not cool, but I understand a problem.

please ask question on how to do that on stackoverflow and maven-site-plugin mail-list, share links with me.

**Comment by alexkravin** [comment]
I found no way to insert javascript or smth else in section name, xdocs are too strict.

I have also asked this question at stackoverflow 2 weeks ago:

http://stackoverflow.com/questions/28752749/maven-site-generation-xdoc-make-anchor-put-link-with-image-in-section-titl

http://ocim.website/28752749/maven-site-generation-xdoc-make-anchor-put-link-with-image-in-section-titl.html

And received no response.

**Comment by romani** [comment]
What about this http://maven.apache.org/plugins/maven-site-plugin/examples/sitedescriptor.html#Inject_xhtml_into_head ?

**Comment by alexkravin** [comment]
I saw this, how does it help with <b>insert javascript or smth else in section name</b>

**Comment by romani** [comment]
you do not need to insert js to section, github wiki pages have no js code at link and it works. Do it the same way.
Example : https://github.com/checkstyle/checkstyle/wiki/How-to-make-a-release
```
<h1>
    <a id="user-content-pre-reqs" class="anchor" aria-hidden="true" href="#pre-reqs"></a>
    Pre-reqs
</h1>
```

**Comment by alexkravin** [comment]
I saw this too, but didn't find a way how to make it as link near the section in xdoc, I'll investigate it again.

**Comment by alexkravin** [comment]
Only section element is allowed in &lt;body&gt;.

Anchors are generated implicitly while parsing as it was asked here:

http://jira.codehaus.org/browse/DOXIA-245

and as we do not specify section's id - parser generates it from section's name attribute.

also see http://jira.codehaus.org/browse/MSITE-167
which describes the similar problem, all these issues are pointing at
http://maven.apache.org/archives/maven-1.x/plugins/xdoc/reference/xdocs.html#Referencing_sections_and_subsections
now; and there's no mention about explicit anchors, furthermore, as you see this site does not have any explicit anchors too and there's no doubt it's maven generated site

**Comment by romani** [comment]
You do not need to change a xdoc files, links a generated by maven correctly.

1) Please generate web site as it generated now
2) work with generated html pages to make required behavior, in the same way as it done at github (there will be no maven, doxia, plugins. ...)
3) show me result

**Comment by alexkravin** [comment]
The problem will be because of xdoc's constraints:

I've tried to make an example with pure html + css:
style:
```

.logo {
  background-image: url(../images/javaguidelink.png);
  display: block;
  width: 20px;
  height: 15px;
  position: absolute;
  left: 180px;
  opacity: 0;
}

a:hover .logo {
  opacity: 1;
}
```

html:
```
<h2><a name="GenericWhitespace" href="#GenericWhitespace"><span class="logo"></span>
</a>GenericWhitespace</h2>
```

There's a need to use &lt;span&gt; tag as we have to modify only part of section's name, the same approach is used on provided github's link in aforesaid.

But we won't have such a possibility in xdoc.

**Comment by alexkravin** [comment]
Included javascript to xdoc but due to some reason it does not work as expected, looks like script isn't called at all

**Comment by alexkravin** [comment]
Resolved.

This is example of one page, if such approach is ok - I'll continue in the same way

http://alexkravin.github.io/anchors/config_annotation.html

**Comment by romani** [comment]
looks good.

1) please put image to right side of Link text , it will looks more natural (in the same grey area as text).

1.1) please place image at the middle by vertical align, (right now image is shown at the top ). It that is complicated, we could changed chain image to bigger.

2) please do the same for all sections (h2 and h3), in other words on that page I expect image to appear for "Description", "Properties", .....

**Comment by alexkravin** [comment]
Moved image to middle alignment and to grey area as relatively positioned element, but in all examples of anchors usage I saw them on the <b>left</b> of text, so I think it's better way.

I found no way of addressing subsections, as variant like:
http://alexkravin.github.io/anchors/config_blocks.html#EmptyBlock#Description
does not work, furthermore, if address like http://alexkravin.github.io/anchors/config_blocks.html#Description
it will reference to the first occurrence on the page, so it won't have any logic.

Updated links in all pages:

http://alexkravin.github.io/anchors/index.html

**Comment by romani** [comment]
>  if address like http://alexkravin.github.io/anchors/config_blocks.html#Description

ok, please open issue against maven site plugin to generate unique names for all sections.

**Comment by romani** [comment]
fixed, will be in in 6.5
